core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -129,6 +129,8 @@ class UserService(protected val ctx: RawlsRequestContext,
 
   import dataSource.dataAccess.driver.api._
 
+  val billingRepository = new BillingRepository(dataSource)
```

Comment:
Minor nit, instead of a direct instantiation, this could added as a parameter to the (admittedly gargantuan) constructor. We are instantiating a billing repository instance during boot [here](https://github.com/broadinstitute/rawls/blob/develop/core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala#L497) which could likely be threaded through the UserService ctor [call](https://github.com/broadinstitute/rawls/blob/develop/core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala#L312)

Reply:
That also allows for better mocking in tests. Right now you'd need to.... override the class? I don't think `spy` would even work, since it's a `val`, not a function.

Reply:
This is the approach I did initially in https://github.com/broadinstitute/rawls/commit/7d22a12f5fb2e29e6024a4b25a7a784db2d0e6ba (discussed inhttps://github.com/broadinstitute/rawls/pull/2046#discussion_r976415528). However, it led to lots of changes because every place UserService is instantiated (tests), I had to pass a BillingRepository. And I didn't see that it added any value given that UserService still takes slickDataSource as an input.

I think we eventually want to move to passing BillingRepository instead of slackDataSource, but we are a ways off from that.

Reply:
And yes, passing it into `UserService` did allow for mocking instead of using the minimal test database. I ended up backing that test change out, but do recognize that's the direction we ultimately want to go.

@blakery if you want to change to passing `BillingRepository` into `UserService` as part of your ticket, go for it!